pkg/controller/status/controller.go

```diff
@@ -131,6 +131,9 @@ func (c *Controller) merge(clusterOperator *configv1.ClusterOperator) *configv1.
 		clusterOperator = newClusterOperator(c.name, nil)
 	}
 
+	// make sure to start a clean status controller
+	c.ctrlStatus.reset()
```

Comment:
@tremes So, after a few tests (and fighting with the e2e tests), it seems that this is the best solution. We keep an instance to the status controller into the controller (controller everywhere) and then it encapsulates the logic to handle the statuses. But, it needs to be reset every time that the merge is executed, otherwise, the operator status would be wrong.

Reply:
yes I am looking at it and it seem we are reseting everything (also the line https://github.com/openshift/insights-operator/pull/498/files#diff-10c33b0f428af9470aef2747e42fc97d97150d338b151e836d9cdebbf4fbef42R149) eveyrtime....

Reply:
Yeh, but in this case it is correct. We are "resetting" 'cause the `merge` method receives the `clusterOperator` when it is called.